shell/platform/linux/fl_binary_messenger.cc

```diff
@@ -311,6 +316,32 @@ static GBytes* send_on_channel_finish(FlBinaryMessenger* messenger,
   return fl_engine_send_platform_message_finish(engine, r, error);
 }
 
+static void resize_channel(FlBinaryMessenger* messenger,
+                           const gchar* channel,
+                           int new_size) {
+  g_autoptr(FlStandardMethodCodec) codec = fl_standard_method_codec_new();
+  g_autoptr(FlValue) args = fl_value_new_list();
+  fl_value_append_take(args, fl_value_new_string(channel));
+  fl_value_append_take(args, fl_value_new_int(new_size));
+  g_autoptr(GBytes) message = fl_method_codec_encode_method_call(
+      FL_METHOD_CODEC(codec), "resize", args, nullptr);
+  fl_binary_messenger_send_on_channel(messenger, kControlChannelName, message,
```

Comment:
Might be worth handling the result here in case there's an error which could be shown with g_warning. Otherwise you'll never know.

Reply:
I'm not sure how to do this. I was not able to find a place where a similar handling is done, can you point out an example?

Reply:
See https://github.com/flutter/engine/blob/533670269c99a8f552edb1e5bf5a17b706e54e5b/shell/platform/linux/fl_text_input_plugin.cc#L164 and https://github.com/flutter/engine/blob/533670269c99a8f552edb1e5bf5a17b706e54e5b/shell/platform/linux/fl_text_input_plugin.cc#L116 for a case where a method call is only completed to check for the error. There's no other cases using `fl_binary_messenger_send_on_channel` directly so in this case you'll need to use `fl_binary_messenger_send_on_channel_finish` and `fl_method_codec_decode_response`.